scout/commands/update/genes.py

```diff
@@ -17,94 +17,170 @@
+    if downloads_folder is None:
+        with tempfile.TemporaryDirectory() as tempdir:
+            try:
+                download_resources(tempdir, api_key, builds)
+            except Exception as ex:
+                LOG.error(ex)
+        fetch_downloaded_resources(resources, tempdir, builds)
+    else:  # If resources have been previosly downloaded, read those file and return their lines
+        fetch_downloaded_resources(resources, downloads_folder, builds)
 
     LOG.warning("Dropping all gene information")
     adapter.drop_genes(build)
-    LOG.info("Genes dropped")
     LOG.warning("Dropping all transcript information")
     adapter.drop_transcripts(build)
-    LOG.info("transcripts dropped")
-
-    hpo_genes = fetch_genes_to_hpo_to_disease()
-
-    if build:
-        builds = [build]
-    else:
-        builds = ["37", "38"]
-
-    hgnc_lines = fetch_hgnc()
-    exac_lines = fetch_exac_constraint()
 
+    # Load genes and transcripts info
     for genome_build in builds:
-        ensembl_genes = fetch_ensembl_genes(build=genome_build)
+        ensembl_gene_res = (
+            resources.get("ensembl_genes_37")
+            if genome_build == "37"
+            else resources.get("ensembl_genes_38")
+        )  # It will be none if everything needs to be downloaded
 
-        # load the genes
+        # Load the genes
         hgnc_genes = load_hgnc_genes(
             adapter=adapter,
-            ensembl_lines=ensembl_genes,
-            hgnc_lines=hgnc_lines,
-            exac_lines=exac_lines,
-            mim2gene_lines=mim_files.get("mim2genes"),
-            genemap_lines=mim_files.get("genemap2"),
-            hpo_lines=hpo_genes,
+            ensembl_lines=ensembl_gene_res,
+            hgnc_lines=resources.get("hgnc_lines"),
+            exac_lines=resources.get("exac_lines"),
+            mim2gene_lines=resources.get("mim2genes"),
+            genemap_lines=resources.get("genemap2"),
+            hpo_lines=resources.get("hpo_genes"),
             build=genome_build,
         )
 
-        ensembl_genes = {}
+        ensembl_genes_dict = {}
         for gene_obj in hgnc_genes:
             ensembl_id = gene_obj["ensembl_id"]
-            ensembl_genes[ensembl_id] = gene_obj
+            ensembl_genes_dict[ensembl_id] = gene_obj
 
-        # Fetch the transcripts from ensembl
-        ensembl_transcripts = fetch_ensembl_transcripts(build=genome_build)
+        # Load the transcripts
+        ensembl_tx_res = (
+            resources.get("ensembl_transcripts_37")
+            if genome_build == "37"
+            else resources.get("ensembl_transcripts_38")
+        )  # It will be none if everything needs to be downloaded
 
-        transcripts = load_transcripts(adapter, ensembl_transcripts, genome_build, ensembl_genes)
+        load_transcripts(adapter, ensembl_tx_res, genome_build, ensembl_genes_dict)
 
     adapter.update_indexes()
 
-    LOG.info("Genes, transcripts and Exons loaded")
```

Comment:
Hm, yes, the load exons is really also an update command. I foresee a unification at a (major?) in the future.

Reply:
Sounds good